### PR TITLE
feat: impl `BatchDelete`

### PR DESCRIPTION
**Pull request by shuiyisong**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?
This pr mainly implements `BatchDelete` on `kv_store`

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
#1249

### CODE REVIEW

File: src/client/src/database.rs [hunk 1]
```diff
@@ -86,6 +86,7 @@ impl Database {
                 catalog: self.catalog.clone(),
                 schema: self.schema.clone(),
                 authorization: self.ctx.auth_header.clone(),
+                dbname: format!("{}-{}", self.catalog, self.schema),
```
Comment (shuiyisong) [inline src/client/src/database.rs:89]: @sunng87 please check this out whether it's desired

**Comment by Fengys123** [comment]
The corresponding `batch delete` API needs to be added in metasrv client crate.

File: src/client/src/database.rs (hunk 1 above)
Comment (sunng87) [reply src/client/src/database.rs:89]: I think I should update my patch first. Here if catalog and schema is provided, we can leave dbname a default empty string. I'm trying to  avoid leaking this concat pattern to client

File: src/meta-srv/src/service/store/memory.rs [hunk 2]
```diff
@@ -163,6 +164,34 @@ impl KvStore for MemStore {
...
+        let mut memory = self.inner.write();
+        let prev_kvs = if prev_kv {
+            keys.into_iter()
+                .map(|key| (key.clone(), memory.remove(&key)))
+                .filter(|(_, v)| v.is_some())
+                .map(|(key, value)| KeyValue {
+                    key,
+                    value: value.unwrap(),
+                })
+                .collect()
```
Comment (MichaelScofield) [inline src/meta-srv/src/service/store/memory.rs:183]:
```suggestion
            keys.into_iter()
                .filter_map(|key| memory.remove(&key).map(|value| KeyValue { key, value }))
                .collect()
```

**Comment by shuiyisong** [comment]
@Fengys123 Rebased, PTAL 😆

File: src/servers/src/auth/user_provider.rs [hunk 3]
```diff
@@ -106,22 +107,40 @@ impl UserProvider for StaticUserProvider {
...
                             }
                             .fail()
-                        }
+                        };
                     }
                     Password::MysqlNativePassword(auth_data, salt) => {
+                        ensure!(
+                            auth_data.len() == 20,
+                            IllegalParamSnafu {
+                                msg: "blank password"
```
Comment (MichaelScofield) [inline src/servers/src/auth/user_provider.rs:141]:
```suggestion
                                msg: "Illegal MySQL native password format, length != 20"
```

**Review by MichaelScofield — APPROVED** [review]

**Review by Fengys123 — APPROVED** [review]
LGTM

**Pull request merged**